roiextractors/extractors/caiman/caimansegmentationextractor.py

```diff
@@ -0,0 +1,151 @@
+import numpy as np
+import h5py
+from lazy_ops import DatasetView
+from ...segmentationextractor import SegmentationExtractor
+from ...extraction_tools import _pixel_mask_extractor
+
+class CaimanSegmentationExtractor(SegmentationExtractor):
+    """
+    This class inherits from the SegmentationExtractor class, having all
+    its funtionality specifically applied to the dataset output from
+    the \'CNMF-E\' ROI segmentation method.
+    """
+    extractor_name = 'CaimanSegmentation'
+    installed = True  # check at class level if installed or not
+    is_writable = False
+    mode = 'file'
+    installation_mesg = ""  # error message when not installed
+
+    def __init__(self, filepath):
+        """
+        Parameters
+        ----------
+        filepath: str
+            The location of the folder containing caiman *.hdmf output file.
+        """
+        SegmentationExtractor.__init__(self)
+        self.filepath = filepath
+        self._dataset_file = self._file_extractor_read()
+        self._roi_response = self._trace_extractor_read('F_dff')
+        self._masks, self._mask_roi_ids, self._mask_ids = self._image_mask_sparse_read()
```

Comment:
Why would `self_mask_roi_ids` be different from `self._roi_ids`?

Reply:
these are different variables. 
1. `self._masks`  is a 1d array containing pixel weights of all pixels in an image_mask across all ROIs. There are two other arrays keeping track two indices: 
1. `self._mask_roi_ids` are ids for the pixel relative to each ROI (flattened). Since there is one location for each this array is of length `self._masks` 
2. `self._mask_ids` index the  `self._masks` and mark the change in ROI. So this is length no_rois. 

`self._image_mask_sparse_read()` just pulls out these 3 arrays.  `self.get_roi_image_masks` has the logic to build a 3d image_mask array out of that

Reply:
I'd rather we fetch these with the necessary code to massage them into the same internal representation as other SegmentationExtractors, rather than keeping the same form as CaImAn under the hood

Reply:
fixed here 87f31df3c08548e722c9396defb3fe1a3587f41c
generating image_masks from the h5py file would be expensive since it requires a reshape operation, can this be done lazily?
Keeping this in mind, I have kept `self.image_masks = None ` in `__init__`

Reply:
reshape cannot be done lazily currently

Reply:
Could you populate self.image_masks in the constructor?

Reply:
would that be necessary? 
image_masks is necessary for every subclass to re-define anyway since its an important attribute